static/js/millennium_diff_table.js

```diff
@@ -0,0 +1,608 @@
+const conformanceURL = 'https://fhir-ehr.sandboxcerner.com/dstu2/0b8a0111-e8e6-4c26-a91c-5069cbc6b1ca/metadata';
+const conformanceHeaders = { Accept: 'application/json+fhir' };
+
+const capabilityStatementURL = 'https://fhir-ehr.sandboxcerner.com/r4/0b8a0111-e8e6-4c26-a91c-5069cbc6b1ca/metadata';
+const capabilityStatementHeaders = { Accept: 'application/fhir+json' };
+
+// Configuration for resources that have been altered from DSTU2 to R4, or extended from Basic
+const resourceConfig = ({
+  combinedResources: [
+    {
+      dstu2Resources: ["Conformance"],
+      r4Resources: ["CapabilityStatement"],
+      notes: "The Conformance resource was renamed to CapabilityStatement in R4."
+    },
+    {
+      dstu2Resources: ["MedicationOrder", "MedicationStatement"],
+      r4Resources: ["MedicationRequest"],
+      notes: "Our implementations of the MedicationOrder and MedicationStatement resources have been implemented into MedicationRequest in R4."
+    },
+    {
+      dstu2Resources: ["ProcedureRequest"],
+      r4Resources: ["ServiceRequest"],
+      notes: "The ProcedureRequest resource was renamed to ServiceRequest in R4."
+    }
+  ],
+  basicResources: [
+    {
+      type: "FinancialTransaction",
+      interaction: [
+        {
+          code: "create"
+        }
+    ],
+    notes: "FinancialTransaction is a custom resource implemented through the Basic resource."
+    }
+  ]
+});
+
+/**
+ * Retreives the configuration for a specific resource by its name.
+ * @param {string} resourceName - The name of the resource to find.
+ * @returns {array} Array of all Basic resource configurations.
+ * @returns {object} The configuration object for the given resource.
+ * @returns {null} No configuration exists for the given resource.
+ */
+function getConfiguration(resourceName) {
+  if (resourceName === "Basic") {
+    return resourceConfig.basicResources;
+  }
+  else {
+    let combinedResources = resourceConfig.combinedResources;
+    for (let i = 0; i < combinedResources.length; i++) {
+      if (combinedResources[i].dstu2Resources.includes(resourceName)) {
+        return combinedResources[i];
+      }
+    }
+  }
+  return null;
+}
+
+/**
+ * Converts a given resource into a new object with the only resource's name and supported actions.
+ * @param {object} resourceObject - The object containing a resource's information.
+ * @returns {object} The new resource object.
+ */
+function buildResource(resourceObject) {
```

Comment:
we'll take a performance hit but i think the method is a lot more readable this way

```
function buildResource(resourceObject) {
  let supportedActions = resourceObject.interaction;

  return {
    resourceName: resourceObject.type,
    readSupported: supportedActions.filter(action => action.code == "read").length > 0,
    searchSupported: supportedActions.filter(action => action.code == "search-type").length > 0,
    createSupported: supportedActions.filter(action => action.code == "create").length > 0,
    updateSupported: supportedActions.filter(action => action.code == "update").length > 0,
    patchSupported: supportedActions.filter(action => action.code == "patch").length > 0,
    deleteSupported: supportedActions.filter(action => action.code == "delete").length > 0
  };
}
```

Reply:
Updated here: c4b951d85ae01dcd47ad911dcf224da107f2adcb